File: frontend/src/types/assistant.ts
export interface AssistantSettings {
  id: number;
  voice: string;
  system_instructions: string;
  greeting_message: string | null;
  template_name: string;
  phone_number: string | null;
  created_at: string;
  updated_at: string;
}

export interface AssistantSettingsUpdate {
  voice?: string;
  system_instructions?: string;
  greeting_message?: string | null;
  template_name?: string;
  phone_number?: string | null;
}

export interface InstructionTemplate {
  id: number;
  name: string;
  description: string;
  default_instructions: string;
  category: string;
}

export interface Voice {
  id: string;
  name: string;
  description: string;
}

File: backend/app/models/assistant.py
"""Assistant settings model."""

from datetime import datetime
from typing import Optional

from sqlmodel import Field, SQLModel


class AssistantSettings(SQLModel, table=True):
    """Persisted assistant configuration."""

    __tablename__ = "assistant_settings"

    id: Optional[int] = Field(default=None, primary_key=True)
    voice: str = Field(default="sage", max_length=50)
    system_instructions: str = Field(default="")
    greeting_message: Optional[str] = Field(default=None)
    template_name: str = Field(default="allgemein", max_length=100)
    phone_number: Optional[str] = Field(default=None, max_length=50, index=True)
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)
